github/actions/testserver/server.go

```diff
@@ -0,0 +1,115 @@
+package testserver
```

Comment:
Extracted the actions server into its own package so it can be used in all tests, didn't use it everywhere yet to try and keep this PR sub 1000 lines 😓